### PR TITLE
Anna dagger2

### PR DESCRIPTION
**Pull request by Anna812**
with the help of Balint, Dagger was implemented to inject Mockserver to LoginActivity

### CODE REVIEW

File: app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/LoginActivity.java [hunk 1]
```diff
@@ -32,6 +35,8 @@
 
     String username, password;
     ReaderApi api;
+    @Inject
+    MockServer mockServer;
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/LoginActivity.java]: @Anna812 the point would be to inject the ReaderApiInterface so the usage will be the same and this class won't change, you just change the the implementation of it

Comment (Anna812) [reply app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/LoginActivity.java]: updated according to your comment

File: app/src/main/java/com/greenfox/fuchsit/zerdareader/server/MockServer.java [hunk 2]
```diff
@@ -3,15 +3,15 @@
...
+import dagger.Provides;
 import retrofit2.Callback;
 import retrofit2.Response;
-import retrofit2.http.Query;
 
 /**
  * Created by Zsuzska on 2017. 01. 20..
  */
-
+@Module
```
Comment (adamgyulavari) [inline app/src/main/java/com/greenfox/fuchsit/zerdareader/server/MockServer.java]: @Anna812 into a specific class: `ReaderApiProvider`

**Comment by adamgyulavari** [comment]
@Anna812 resolve conflicts :)

**Comment by Anna812** [comment]
added ApiProvider, resolved conflict, deleted duplication

**Pull request merged**